test/purview_test.tsx

```diff
@@ -1907,6 +1908,86 @@ test("reconnect early", async () => {
   })
 })
 
+test("simulate a page refresh", async () => {
```

Comment:
I will admit that this is pretty complex and the only way I could truly verify it worked is with breadcrumbs while running the test. Using breakpoints would cause the race condition to not be seen